eclipse-collections-api/src/main/java/org/eclipse/collections/api/MutableIterable.java

```diff
@@ -0,0 +1,222 @@
+ */
+
+package org.eclipse.collections.api;
+
+import org.eclipse.collections.api.block.function.Function;
+import org.eclipse.collections.api.block.function.Function0;
+import org.eclipse.collections.api.block.function.Function2;
+import org.eclipse.collections.api.block.function.primitive.DoubleFunction;
+import org.eclipse.collections.api.block.function.primitive.FloatFunction;
+import org.eclipse.collections.api.block.function.primitive.IntFunction;
+import org.eclipse.collections.api.block.function.primitive.LongFunction;
+import org.eclipse.collections.api.block.predicate.Predicate;
+import org.eclipse.collections.api.block.predicate.Predicate2;
+import org.eclipse.collections.api.block.procedure.Procedure;
+import org.eclipse.collections.api.block.procedure.Procedure2;
+import org.eclipse.collections.api.map.MutableMap;
+import org.eclipse.collections.api.map.MutableMapIterable;
+import org.eclipse.collections.api.map.primitive.MutableObjectDoubleMap;
+import org.eclipse.collections.api.map.primitive.MutableObjectLongMap;
+import org.eclipse.collections.api.multimap.MutableMultimap;
+import org.eclipse.collections.api.ordered.OrderedIterable;
+import org.eclipse.collections.api.partition.PartitionMutableIterable;
+import org.eclipse.collections.api.tuple.Pair;
+
+public interface MutableIterable<T> extends RichIterable<T>
+{
+    @Override
+    MutableIterable<T> tap(Procedure<? super T> procedure);
+
+    @Override
+    MutableIterable<T> select(Predicate<? super T> predicate);
+
+    @Override
+    <P> MutableIterable<T> selectWith(Predicate2<? super T, ? super P> predicate, P parameter);
+
+    @Override
+    MutableIterable<T> reject(Predicate<? super T> predicate);
+
+    @Override
+    <P> MutableIterable<T> rejectWith(Predicate2<? super T, ? super P> predicate, P parameter);
+
+    @Override
+    PartitionMutableIterable<T> partition(Predicate<? super T> predicate);
+
+    @Override
+    <P> PartitionMutableIterable<T> partitionWith(Predicate2<? super T, ? super P> predicate, P parameter);
+
+    @Override
+    <S> MutableIterable<S> selectInstancesOf(Class<S> clazz);
+
+    @Override
+    <V> MutableIterable<V> collect(Function<? super T, ? extends V> function);
+
+    /*
+    @Override
+    MutableBooleanIterable collectBoolean(BooleanFunction<? super T> booleanFunction);
+
+    @Override
+    MutableByteIterable collectByte(ByteFunction<? super T> byteFunction);
+
+    @Override
+    MutableCharIterable collectChar(CharFunction<? super T> charFunction);
+
+    @Override
+    MutableDoubleIterable collectDouble(DoubleFunction<? super T> doubleFunction);
+
+    @Override
+    MutableFloatIterable collectFloat(FloatFunction<? super T> floatFunction);
+
+    @Override
+    MutableIntIterable collectInt(IntFunction<? super T> intFunction);
+
+    @Override
+    MutableLongIterable collectLong(LongFunction<? super T> longFunction);
+
+    @Override
+    MutableShortIterable collectShort(ShortFunction<? super T> shortFunction);
+    */
```

Comment:
Thanks for clarifying. I don't think they need to be in the same commit. 

For these comments, maybe just put TODO so that these can be easily found later on?